os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -151,3 +217,373 @@ uint64_t imxrt_pit_getlifetimetimercount(PIT_Type *base)
+
+	imxrt_pit_init(pit->base, config);
+	imxrt_pit_start_lifetimetimercount(pit->base);
+
+	return pit;
+}
+
+void imxrt_pit_setisr(struct imxrt_pit_chipinfo_s *pit, xcpt_t handler, void *arg)
+{
+	irq_attach(pit->irq_id, handler, arg);
+
+	up_enable_irq(pit->irq_id);
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_handler
+ *
+ * Description:
+ *   timer interrupt handler
+ *
+ * Input Parameters:
+ *
+ * Returned Values:
+ *
+ ****************************************************************************/
+static int imxrt_pit_handler(int irq, void *context, void *arg)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)arg;
+	struct imxrt_pit_chipinfo_s *pit = priv->pit;
+	uint32_t next_interval_us = 0;
+	uint32_t freq;
+	uint64_t ticks;
+
+	imxrt_pit_clearstatusflags(pit->base, kPIT_Chnl_3, kPIT_TimerFlag);
+	if (ch_status[kPIT_Chnl_3])
+		imxrt_pit_clearstatusflags(pit->base, kPIT_Chnl_2, kPIT_TimerFlag);
+
+	if (priv->callback(&next_interval_us, priv->arg)) {
+		if (next_interval_us > 0) {
+			freq = imxrt_clock_getperclkfreq();
+			ticks = ((uint64_t)freq * (uint64_t)next_interval_us) / USEC_PER_SEC;
+			if (ch_status[kPIT_Chnl_3]) {
+				uint32_t ch_ticks1 = (uint32_t)freq /USEC_PER_SEC;
+				uint32_t ch_ticks2 = (uint32_t)next_interval_us;
+				imxrt_pit_settimerperiod(priv->pit->base, kPIT_Chnl_3, ch_ticks1);
+				imxrt_pit_settimerperiod(priv->pit->base, kPIT_Chnl_2, ch_ticks2);
+				}
+		}
+	} else {
+		imxrt_pit_stop((struct timer_lowerhalf_s *)priv);
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_start
+ *
+ * Description:
+ *   Start the timer, resetting the time to the current timeout,
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int imxrt_pit_start(struct timer_lowerhalf_s *lower)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
+
+	if (!priv->started) {
+		if (priv->callback != NULL) {
+			imxrt_pit_enableinterrupts(priv->pit->base, kPIT_Chnl_3, kPIT_TimerInterruptEnable);
+		}
+
+		imxrt_pit_setisr(priv->pit, imxrt_pit_handler, priv);
+		imxrt_pit_starttimer(priv->pit->base, kPIT_Chnl_3);
+
+		priv->started = true;
+#if defined(FSL_FEATURE_PIT_HAS_CHAIN_MODE) && FSL_FEATURE_PIT_HAS_CHAIN_MODE
+		if (ch_status[kPIT_Chnl_3])
+			imxrt_pit_starttimer(priv->pit->base, kPIT_Chnl_2);
+
+		tmrvdbg("ch_status:%d/%d/%d/%d\n", ch_status[0], ch_status[1], ch_status[2], ch_status[3]);
+#endif
+		tmrvdbg("PIT Timer started, callback %s\n", priv->callback == NULL ? "none" : "set");
+		return OK;
+	}
+
+	/* Return EBUSY to indicate that the timer was already running */
+

```

Comment:
We can remove these extra lines between comment and code

Reply:
Done